libs/client/cornflow_client/core/instance.py

```diff
@@ -76,8 +75,28 @@ def launch_all_checks(self) -> Dict[str, Union[List, Dict]]:
         and the list/dict of errors / warnings as value.
 
         It will only return those checks that return a non-empty list/dict.
+        If a check method raises an exception, it will be caught and a generic error
+        message will be added to the checks dictionary.
         """
-        check_methods = {m: getattr(self, m)() for m in self.get_check_methods()}
+        check_method_names = self.get_check_methods()
+        check_methods = {}
+
+        # Execute each check method and catch any exceptions
```

Comment:
this code is duplicated between instance and experiment. Maybe it should be in experimentcore